web/react/components/sidebar.jsx

```diff
@@ -197,6 +197,19 @@ module.exports = React.createClass({
             if (UserStore.getCurrentId() === msg.user_id) {
                 AsyncClient.getChannels(true);
             }
+        } else if(msg.action === "user_removed") {
+            if(msg.user_id === UserStore.getCurrentId()) {
+                AsyncClient.getChannels(true);
+
+                if(msg.props.channel_id === ChannelStore.getCurrentId() && $('#removed_from_channel').length > 0) {
+                    var channelName = ChannelStore.getCurrent().display_name;
+                    var curUser = UserStore.getProfile(msg.props.remover).username;
+                    $('#removed_from_channel').find('.modal-title').text("Removed from " + channelName);
+                    $('#removed_from_channel').find('.modal-body').children().text(curUser + " removed you from " + channelName);
+
+                    $('#removed_from_channel').modal('show');
```

Comment:
Yes, the modal could listen for the user_removed event itself, that would be fine. 

If were to use AppDispatcher every time we opened a modal then we would have a large number of listeners for a single event. Basically every modal would have to listen for some 'modal_changed' event on the AppDispatcher and then update it's state based on that. The default limit to the number of listeners is 10, so we would have to update that each time we added a new modal and every modal would need to check "is this event for me specifically?". In my opinion, it's much better to just pass the modal data in this case through the data-* attributes like Bootstrap suggests. For special cases like this where we can't do that, I'd rather not have them go through the AppDispatcher either way for the same reason as not doing all modals, since the number of special cases like this is bound to grow.